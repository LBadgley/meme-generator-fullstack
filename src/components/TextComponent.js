import React from 'react';
import PropTypes from 'prop-types';

export default function TextPicker({ topText, bottomText, handleChange }) {
  return (
    <>
      <p>Top Text:</p>
      <input name="topText" type="text" value={topText} onChange={handleChange} />
      <p>Bottom text:</p>
      <input name="bottomText" type="text" value={bottomText} onChange={handleChange} />
    </>
  );
}

TextPicker.propTypes = {
  topText: PropTypes.string.isRequired,
  bottomText: PropTypes.string.isRequired,
  handleChange: PropTypes.func.isRequired
};
