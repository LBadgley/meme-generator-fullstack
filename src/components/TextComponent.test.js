import React from 'react';
import { shallow } from 'enzyme';
import TextComponent from './TextComponent';

describe('text component test', () => {
  it('renders the text component', () => {
    const wrapper = shallow(<TextComponent />);
    expect(wrapper).toMatchSnapshot();
  });
});
