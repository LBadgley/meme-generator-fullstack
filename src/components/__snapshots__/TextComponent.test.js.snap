// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`text component test renders the text component 1`] = `
<Fragment>
  <p>
    Top Text:
  </p>
  <input
    name="topText"
    type="text"
  />
  <p>
    Bottom text:
  </p>
  <input
    name="bottomText"
    type="text"
  />
</Fragment>
`;
